ethstorage/storage_manager.go

```diff
@@ -26,22 +32,27 @@ type Il1Source interface {
 	GetKvMetas(kvIndices []uint64, blockNumber int64) ([][32]byte, error)
 
 	GetStorageLastBlobIdx(blockNumber int64) (uint64, error)
+
+	HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error)
 }
 
 // StorageManager is a higher-level abstract of ShardManager which provides multi-thread safety to storage file read/write
 // and a consistent view of most-recent-finalized L1 block.
 type StorageManager struct {
 	shardManager      *ShardManager
-	mu                sync.Mutex // protect localL1 and underlying blob read/write
-	localL1           int64      // local view of most-recent-finalized L1 block
+	mu                sync.Mutex // protect lastKvIdx, shardManager and blobMeta read/write state
+	lastKvIdx         uint64     // lastKvIndex in the most-recent-finalized L1 block
 	l1Source          Il1Source
 	DownloadThreadNum int
+	metaMapMu         sync.Mutex // this is only for blobMetas writing thread safe in downloadMetaInParallel
```

Comment:
Both read/write need the lock, and we should comment that we allow lock sequence like mu -> metaMapMu, but never metaMapMu -> mu to avoid dead lock.